### PR TITLE
Change: Always build baseset metadata, instead of including it in the repo.

### PR DESCRIPTION
**Pull request by PeterN**
Baseset metadata files are usually out of date as they are not updated when translations are changed.

This also adds a script and project files to generate these files within MSVC.

### CODE REVIEW

File: media/baseset/translations.vbs [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+
+	Set WSO = Nothing
+End Function
+
+' Simple insertion sort, copied from translations.awk
+Sub ISort(a)
+	Dim i, j, n, hold
+	n = UBound(a)
+
+	For i = 1 To n
```
Comment (glx22) [inline media/baseset/translations.vbs]: arrays are 0 based

File: projects/openttd_vs140.vcxproj [hunk 2]
```diff
@@ -1361,6 +1361,10 @@
       <Project>{1a2b3c5e-1c23-41a5-9c9b-acba2aa75fec}</Project>
       <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
     </ProjectReference>
+		<ProjectReference Include="basesets_vs140.vcxproj">
```
Comment (glx22) [inline projects/openttd_vs140.vcxproj]: wrong alignment

File: projects/openttd_vs140.vcxproj.in [hunk 3]
```diff
@@ -365,6 +365,10 @@
       <Project>{1a2b3c5e-1c23-41a5-9c9b-acba2aa75fec}</Project>
       <ReferenceOutputAssembly>false</ReferenceOutputAssembly>
     </ProjectReference>
+		<ProjectReference Include="basesets_vs140.vcxproj">
```
Comment (glx22) [inline projects/openttd_vs140.vcxproj.in]: wrong alignment

File: media/baseset/translations.vbs [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+	' Ensure only complete string matches
+	str_id = str_id & " "
+
+	Set folder = FSO.GetFolder(langpath)
+
+	Dim output()
+	ReDim output(folder.Files.Count)
+
+	For Each file In folder.Files
+		If UCase(fso.GetExtensionName(file.Name)) = "TXT" Then
```
Comment (glx22) [inline media/baseset/translations.vbs]: FSO ;)

**Review by glx22 — DISMISSED** [review]
Where is openttd_vs140.sln ?

**Comment by PeterN** [comment]
Missed that. Have added it now, but haven't got a version that'll open it.

**Comment by PeterN** [comment]
Hmm, thinking about it, it probably needs to depend on all the language files, not just english.txt. Can't see how to do that, replacing english.txt with *.txt doesn't work at least!

**Comment by glx22** [comment]
Tried VS2015, works

**Comment by nielsmh** [comment]
> 
> 
> Hmm, thinking about it, it probably needs to depend on all the language files, not just english.txt. Can't see how to do that, replacing english.txt with *.txt doesn't work at least!

I think you'd need, at least for MSBuild, a dummy-output generated from all the language inputs, and then you can depend other things on that dummy-output.
If it was within the same MSBuild project there would be better solutions, but cross-project having a dummy-file generated is probably the simplest.

**Comment by PeterN** [comment]
Latest push adds code to build a list of language files and include it in the dependences. Seems to work for me with MSVC 2017.

**Comment by glx22** [comment]
Seems to work with MSVC 2015 and MSYS2

**Comment by PeterN** [comment]
I guess the final question now is... is this change a good thing to do? Seeing as eints likely doesn't have the build environment to update these files, this seems the best way to me.

**Comment by nielsmh** [comment]
Yeah, IMO this is good. It's less prebuilt files to host in the repository. If it can reasonably be generated during build it should.

**Review by nielsmh — APPROVED** [review]

**Pull request merged**